plugin/trino-teradata/src/main/java/io/trino/plugin/teradata/TeradataClient.java

```diff
@@ -0,0 +1,1431 @@
+            case Types.CHAR:
+                return Optional.of(charColumnMapping(typeHandle.requiredColumnSize(), deriveCaseSensitivity(typeHandle.caseSensitivity().orElse(null))));
+            case Types.VARCHAR:
+                // see prior note on trino case sensitivity
+                return Optional.of(varcharColumnMapping(typeHandle.requiredColumnSize(), deriveCaseSensitivity(typeHandle.caseSensitivity().orElse(null))));
+            case Types.CLOB:
+                return Optional.of(ColumnMapping.sliceMapping(
+                        createUnboundedVarcharType(),
+                        (resultSet, columnIndex) -> utf8Slice(resultSet.getString(columnIndex)),
+                        varcharWriteFunction(),
+                        DISABLE_PUSHDOWN));
+            case Types.BINARY:
+            case Types.VARBINARY:
+            case Types.BLOB:
+                // trino only has varbinary
+                return Optional.of(varbinaryColumnMapping());
+            case Types.DATE:
+                return Optional.of(dateColumnMappingUsingLocalDate());
+            case Types.TIME:
+                return Optional.of(timeColumnMapping(typeHandle.requiredDecimalDigits()));
+            case Types.TIMESTAMP:
+                return Optional.of(timestampColumnMapping(TimestampType.createTimestampType(typeHandle.requiredDecimalDigits())));
+        }
+
+        if (getUnsupportedTypeHandling(session) == CONVERT_TO_VARCHAR) {
+            return mapToUnboundedVarchar(typeHandle);
+        }
+
+        return Optional.empty();
+    }
+
+    /**
+     * Maps a JDBC NUMERIC/DECIMAL type to a Trino DecimalType column mapping.
+     * Handles precision and scale constraints.
+     *
+     * @param typeHandle the JDBC type handle
+     * @return Optional containing the ColumnMapping for the decimal type
+     */
+    private Optional<ColumnMapping> numberMapping(JdbcTypeHandle typeHandle)
+    {
+        int precision = typeHandle.requiredColumnSize();
+        int scale = typeHandle.requiredDecimalDigits();
+        if (precision > Decimals.MAX_PRECISION) {
+            // this will trigger for number(*) as precision is 40
+            return Optional.of(decimalColumnMapping(createDecimalType(Decimals.MAX_PRECISION, scale)));
+        }
+        return Optional.of(decimalColumnMapping(createDecimalType(precision, scale)));
+    }
+
+    /**
+     * This connector is read-only and does not support writing to Teradata.
+     * This method always throws a NOT_SUPPORTED exception.
+     *
+     * @param session connector session
+     * @param type Trino type for the column
+     * @return never returns normally
+     * @throws TrinoException always thrown indicating unsupported operation
+     */
+    @Override
+    public WriteMapping toWriteMapping(ConnectorSession session, Type type)
+    {
+        return switch (type) {
+            case Type t when t.equals(jsonType) -> WriteMapping.sliceMapping("JSON", typedVarcharWriteFunction());
```

Comment:
Avoid abbreviations. https://github.com/trinodb/trino/blob/master/.github/DEVELOPMENT.md#avoid-abbreviations

Reply:
addressed in new PR https://github.com/trinodb/trino/pull/26731